### PR TITLE
Safetensors support

### LINKED ISSUE #186: Add support for loading models via safetensors
Thanks a lot for the awesome project!
Loading models through .bin executables are inherently risky and can lead to malicious code execution. Please consider adding support for loading the [safetensors variant](https://github.com/huggingface/safetensors) of the models provided.

If this is something that you would be interested in, I'd be happy to help out with a PR!

### PR DESCRIPTION
**Pull request by vrdn-23**
This enables support for loading the gliner model using the safetensors variant and also saving the model into a safe tensors format if specified. 

Fixes #186

### CODE REVIEW

File: gliner/model.py [hunk 1]
```diff
@@ -781,17 +790,23 @@ def _from_pretrained(
...
             ) and resize_token_embeddings and not config.labels_encoder:
                 gliner.resize_token_embeddings(add_tokens=add_tokens)
-            state_dict = torch.load(model_file, map_location=torch.device(map_location))
+            if model_file.endswith("safetensors"):
+                state_dict = {}
+                with safe_open(model_file, framework="pt", device=map_location) as f:
+                    for key in f.keys():
+                        state_dict[key] = f.get_tensor(key)
+            else:
+                state_dict = torch.load(model_file, map_location=torch.device(map_location), weights_only=True)
```
Comment (vrdn-23) [inline gliner/model.py:809]: `weights_only` warning added to suppress this warning
```
FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  state_dict = torch.load(model_file, map_location=torch.device(map_location))
  ```

File: gliner/model.py [hunk 2]
```diff
@@ -640,7 +641,7 @@ def prepare_state_dict(self, state_dict):
         """
         new_state_dict = {}
         for key, tensor in state_dict.items():
-            key = re.sub("_orig_mod\.", "", key)
+            key = re.sub(r"_orig_mod\.", "", key)
```
Comment (vrdn-23) [inline gliner/model.py:644]: Added regex matching for valid expression of dot
https://stackoverflow.com/questions/13989640/regular-expression-to-match-a-dot

File: gliner/model.py [hunk 3]
```diff
@@ -6,13 +6,14 @@
 from pathlib import Path
 from typing import Dict, List, Optional, Union
 
-import numpy as np
```
Comment (vrdn-23) [inline gliner/model.py:9]: Removed unused import

**Comment by urchade** [comment]
Any idea how to fix this ? @vrdn-23

**Pull request merged**